commands/src/main/java/org/jclouds/karaf/commands/table/internal/ScriptEngineShellTable.java

```
@@ -26,18 +26,14 @@
  * A shell table implementation that works with groovy expressions.
  */
 public class ScriptEngineShellTable<D extends Object> extends BasicShellTable<D> {
-
-  private final String engine;
-  private final ScriptEngineManager scriptEngineFactory = new ScriptEngineManager();
   private final ScriptEngine scriptEngine;
 
   /**
    * Constructor
    * @param engine
    */
-  public ScriptEngineShellTable(String engine) {
-    this.engine = engine;
-    this.scriptEngine = scriptEngineFactory.getEngineByName(engine);
+  public ScriptEngineShellTable(ScriptEngineManager scriptEngineManager, String engine) {
+    this.scriptEngine = scriptEngineManager.getEngineByName(engine);
```

We'd better fail here instead of silently returning null?